npm/angular/src/mount.ts

```diff
@@ -104,74 +132,84 @@ function bootstrapModule<T extends object> (
 /**
  * Initializes the TestBed
  *
- * @param {Type<T>} component Angular component being mounted
- * @param {TestBedConfig} config TestBed configuration passed into the mount function
- * @returns {TestBed} TestBed
+ * @param {Type<T> | string} component Angular component being mounted or its template
+ * @param {MountConfig} config TestBed configuration passed into the mount function
+ * @returns {Type<T>} componentFixture
  */
-function initTestBed<T extends object> (
-  component: Type<T>,
-  config: TestBedConfig<T>,
-): TestBed {
+function initTestBed<T> (
+  component: Type<T> | string,
+  config: MountConfig<T>,
+): Type<T> {
   const { providers, ...configRest } = config
 
-  const testBed: TestBed = getTestBed()
+  const componentFixture = createComponentFixture(component) as Type<T>
 
-  testBed.resetTestEnvironment()
-
-  testBed.initTestEnvironment(
-    BrowserDynamicTestingModule,
-    platformBrowserDynamicTesting(),
-    {
-      teardown: { destroyAfterEach: false },
-    },
-  )
-
-  testBed.configureTestingModule({
-    ...bootstrapModule(component, configRest),
+  TestBed.configureTestingModule({
+    ...bootstrapModule(componentFixture, configRest),
   })
 
   if (providers != null) {
-    testBed.overrideComponent(component, {
+    TestBed.overrideComponent(componentFixture, {
       add: {
         providers,
       },
     })
   }
 
-  return testBed
+  return componentFixture
+}
+
+@Component({ selector: 'cy-wrapper-component', template: '' })
+class WrapperComponent { }
+
+/**
+ * Returns the Component if Type<T> or creates a WrapperComponent
+ *
+ * @param {Type<T> | string} component The component you want to create a fixture of
+ * @returns {Type<T> | WrapperComponent}
+ */
+function createComponentFixture<T> (
+  component: Type<T> | string,
+): Type<T | WrapperComponent> {
+  if (typeof component === 'string') {
+    TestBed.overrideTemplate(WrapperComponent, component)
```

Comment:
Just to clarify - this is the main part of the PR for the feature, that will trigger the Angular compiler to compile the template?

Reply:
Yes that is correct! We basically accept the template as a string and then override a placeholder `WrapperComponent`'s template with the string passed in, and then make sure the rest of the work we do to handle mount just works as is downstream with the `WrapperComponent`